luigi/contrib/kubernetes.py

```diff
@@ -157,18 +157,35 @@ def max_retrials(self):
         """
         return self.kubernetes_config.max_retrials
 
+    @property
+    def delete_on_success(self):
+        """
+        Delete the Kubernetes workload if the job has ended successfully.
+        """
+        return True
+
+    @property
+    def print_pod_logs_on_exit(self):
+        """
+        Fetch and print the pod logs once the job is completed.
+        """
+        return True
+
     def __track_job(self):
         """Poll job status while active"""
-        while (self.__get_job_status() == "running"):
-            self.__logger.debug("Kubernetes job " + self.uu_name
-                                + " is still running")
+        while not self.__verify_job_has_started():
+            time.sleep(1)
+        self.__print_kubectl_hints()
+        while True:
```

Comment:
Having while(true) kind of loops is a bad programming practice. I suggest to move the return conditions in the while statement.